### PR TITLE
feat: add theme toggle

### LINKED ISSUE #61: Add light / dark mode toggle

### PR DESCRIPTION
**Pull request by LBBO**
Only a draft until @ricardo-osorio-santos adds the sun rays to the SVG and I can animate them. Closes #61

### CODE REVIEW

**Comment by mszekiel** [comment]
Video doesn't change immediately, do you know if we can fix it? Maybe we could switch to image first? Also the switcher animation is a bit off with not all elements animated same way and not hidden fully (se to transparent after hidden?).
image

**Comment by mszekiel** [comment]
https://github.com/ory/summit-platform/assets/12242002/83c453e1-b151-4507-838e-b49037449985

File: src/assets/icon/ThemeToggleIcon.css
Comment (mszekiel) [inline src/assets/icon/ThemeToggleIcon.css:1]: For simplicity you can put it into svg into `<style>` so it's all together

Comment (mszekiel) [inline src/assets/icon/ThemeToggleIcon.css:1]: For simplicity you can put it into svg into <style> so it's all together

**Review by mszekiel — CHANGES_REQUESTED** [review]

**Comment by mszekiel** [comment]
Also IMO the animation is bit too fast, so you do not see it at all.

**Comment by jfcurran** [comment]
from what i have seen this works very well - switching of elements is near instant. I agree that the animation is very fast, but its not bad. What's missing to get this shipped? good job

**Comment by LBBO** [comment]
When a non-designer tackles a design issue 😂 

image

**Comment by LBBO** [comment]
The image was already used as a fallback until the video finished loading. However, on slow connections, that could take a while. I've now added preload tags so the browser immediately loads both images, which actually seems to also be good for preventing LCP issues in #60. I'm a bit unsure if perhaps `prefetch` might be better here since one of the images isn't needed immediately and it could save 200KB during initial page load, but the other image is needed immediately and so I kept it as preload for the moment. WDYT?

**Review by Benehiko — APPROVED** [review]
Looks great!

**Review by Benehiko — APPROVED** [review]
LGTM

**Pull request merged**